Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/BaseAutonomous.java

```diff
@@ -0,0 +1,144 @@
+    int LEFT = 1;
+    int MIDDLE = 2;
+    int RIGHT = 3;
+    AprilTagDetection tagOfInterest = null;
+
+    static final double HOLD_ARM_AT_MID_OR_LOW_POS_POWER = 0.005;
+    static final double HOLD_ARM_AT_GROUND_POS_POWER = 0.01;
+    static final double ARM_RAISE_POWER = 1.0 / 400.0;
+
+    public void initializeAuto() {
+        initializeHardware();
+        //grabberServo.setPosition(GRABBER_HALF_CLOSED);
+
+        int cameraMonitorViewId = hardwareMap.appContext.getResources().getIdentifier("cameraMonitorViewId", "id", hardwareMap.appContext.getPackageName());
+        camera = OpenCvCameraFactory.getInstance().createWebcam(hardwareMap.get(WebcamName.class, "Webcam 1"), cameraMonitorViewId);
+        aprilTagDetectionPipeline = new AprilTagDetectionPipeline(tagSize, fx, fy, cx, cy);
+
+        camera.setPipeline(aprilTagDetectionPipeline);
+        camera.openCameraDeviceAsync(new OpenCvCamera.AsyncCameraOpenListener() {
+            @Override
+            public void onOpened() {
+                camera.startStreaming(CAMERA_WIDTH_PIXELS, CAMERA_HEIGHT_PIXELS, OpenCvCameraRotation.UPRIGHT);
+            }
+
+            @Override
+            public void onError(int errorCode) { }
+        });
+
+        telemetry.setMsTransmissionInterval(50);
+    }
+
+    /**
+     * Detects current tag and gives telemetry of tag ID
+     */
+    public void detectAprilTag() {
+        ArrayList<AprilTagDetection> currentDetections = aprilTagDetectionPipeline.getLatestDetections();
+
+        if (currentDetections.size() != 0) {
+            boolean tagFound = false;
+
+            for (AprilTagDetection tag : currentDetections) {
+                if (tag.id == LEFT || tag.id == MIDDLE || tag.id == RIGHT) {
+                    tagOfInterest = tag;
+                    tagFound = true;
+                    break;
+                }
+            }
+
+            if (tagFound) {
+                telemetry.addLine("Tag of interest is in sight!\n\nLocation data:");
+                tagToTelemetry(tagOfInterest);
+            } else {
+                telemetry.addLine("Don't see tag of interest :(");
+
+                if (tagOfInterest == null) {
+                    telemetry.addLine("(The tag has never been seen)");
+                } else {
+                    telemetry.addLine("\nBut we HAVE seen the tag before; last seen at:");
+                    tagToTelemetry(tagOfInterest);
+                }
+            }
+        }
+        telemetry.update();
+        sleep(20);
+    }
+
+    public void updateTelemetryAfterStart() {
+        if (tagOfInterest != null) {
+            telemetry.addLine("Tag snapshot:\n");
+            tagToTelemetry(tagOfInterest);
+            telemetry.update();
+        } else {
+            telemetry.addLine("No tag snapshot available, it was never sighted during the init loop :(");
+            telemetry.update();
+        }
+    }
+
+    void tagToTelemetry(AprilTagDetection detection) {
```

Comment:
Is this public or private (or something else)? All methods should be explicitly marked.